ios/brave-ios/Sources/BraveWallet/Crypto/Stores/NetworkStore.swift

```diff
@@ -193,8 +193,11 @@ public class NetworkStore: ObservableObject, WalletObserverStore {
     isForOrigin: Bool
   ) async -> SetSelectedChainError? {
     let allAccounts = await keyringService.allAccounts()
-    let allAccountsForNetworkCoin = allAccounts.accounts.filter { $0.coin == network.coin }
-    if allAccountsForNetworkCoin.isEmpty {
+    let allAccountsForNetwork = allAccounts.accounts.filter { account in
```

Comment:
we need the same check in `AssetDetailStore`'s `allAccountsForTokenCoin`

Reply:
Fixed in https://github.com/brave/brave-core/pull/22799/commits/ab0af90a903d47f8a3e2515f05e0946794c431af & updated test plan to include Asset Detail view.

Deposit is v1.66+, so opened a separate issue to make sure correct accounts are selected & displayed in Deposit:
https://github.com/brave/brave-browser/issues/37123